packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/CheckoutConfirm/template.success.tsx

```diff
@@ -86,16 +172,35 @@ const Success: React.FC<InjectedFormProps<{}, Props> & Props> = (props) => {
       ? moment.duration(props.withdrawLockCheck.lockTime, 'seconds').days()
       : 3
 
+  const cardDetails =
+    (requiresTerms && props.cards.filter((card) => card.id === paymentMethodId)[0]) || null
+
+  const isCardPayment = requiresTerms && cardDetails
+
+  const fee = props.order.fee ? props.order.fee : props.quote.fee
+  const totalAmount = fiatToString({
+    unit: counterCurrency,
+    value: counterAmount
+  })
+  const purchase = Number(counterAmount) * 100 - Number(fee)
+
   useEffect(() => {
     if (!requiresTerms) {
-      return setAcceptTerms(true)
+      setAcceptTerms(true)
     }
-  }, [])
+  }, [requiresTerms])
 
   const handleCancel = () => {
     props.simpleBuyActions.cancelSBOrder(props.order)
   }
 
+  const paymentPartnerButton =
+    paymentPartner === BankPartners.YAPILY ? (
+      <FormattedMessage id='copy.next' defaultMessage='Next' />
+    ) : (
+      `${orderType === 'BUY' ? 'Buy' : 'Sell'} ${baseAmount} ${baseCurrency}`
```

Comment:
i think there is an enum to compare to, something like

`orderType === ordertype.BUY`